tools/ci_connector_ops/ci_connector_ops/pipelines/connectors_ci.py

```diff
@@ -0,0 +1,154 @@
+    format_check_status: StepStatus = await tests.check_format(format_check_container)
+
+    install_container: Container = build_context.pipeline(f"{connector.technical_name} - Install connector")
+    installed_connector: Container = await builds.install(connector_ci_client, install_container, extras=["tests", "dev", "main"])
+
+    unit_test_container: Container = installed_connector.pipeline(f"{connector.technical_name} - Unit tests")
+    unit_tests_status: StepStatus = await tests.run_unit_tests(unit_test_container)
+
+    integration_test_container: Container = installed_connector.pipeline(f"{connector.technical_name} - Integration tests")
+    integration_tests_status: StepStatus = await tests.run_integration_tests(integration_test_container)
+
+    build_dev_image_client = connector_ci_client.pipeline(f"{connector.technical_name} - Build dev image")
+    _, connector_image_short_id = await builds.build_dev_image(build_dev_image_client, connector, exclude=[".venv"])
+
+    if gsm_credentials:
+        write_connector_secrets_to_local_storage(connector, gsm_credentials)
+
+    acceptance_tests_container: Container = build_dev_image_client.pipeline(f"{connector.technical_name} - Acceptance tests")
+    # The connector_image_short_id  is used as a cache buster. If it changed the acceptance tests will run.
+    acceptance_tests_status: StepStatus = await tests.run_acceptance_tests(
+        acceptance_tests_container, connector, connector_image_short_id, "airbyte/connector-acceptance-test:dev"
+    )
+
+    # TODO run QA checks: this should probably be done inside a dagger container to benefit from caching?
+    # TODO Upload modified secrets to GSM
+    pipeline_logger.info(f"Format -> {format_check_status}")
+    pipeline_logger.info(f"Unit tests -> {unit_tests_status}")
+    pipeline_logger.info(f"Integration tests -> {integration_tests_status}")
+    pipeline_logger.info(f"Acceptance tests -> {acceptance_tests_status}")
+
+
+async def run_connectors_test_pipelines(connectors: List[Connector], gsm_credentials: str):
+    """Runs a CI pipeline for all the connectors passed.
+
+    Args:
+        connectors (List[Connector]): List of connectors for which a CI pipeline needs to be run.
+        gsm_credentials (str): The GSM credentials to read/write connectors' secrets.
+    """
+    config = dagger.Config(log_output=sys.stdout)
+
+    async with dagger.Connection(config) as dagger_client:
+        for connector in connectors:
+            # We scoped this POC only for python and low-code connectors
+            if connector.language in [ConnectorLanguage.PYTHON, ConnectorLanguage.LOW_CODE]:
+                await run_connector_test_pipelines(dagger_client, connector, gsm_credentials)
+            else:
+                logger.warning(f"Not running test pipeline for {connector.technical_name} as it's not a Python or Low code connector")
```

Comment:
Is this loop synchronous or do we run all tests in parallel?  What if we wanted to throttle these in batches of N?

Reply:
I changed it following @helderco suggestion. This loop is non-blocking now.